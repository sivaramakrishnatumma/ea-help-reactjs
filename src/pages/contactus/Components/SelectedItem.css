.selected-item-container {
  padding-left: 100%;
  padding-right: 100%;
  padding-bottom: 30px;
  margin-left: -100%;
  margin-right: -100%;
  margin-top: 27px;
  overflow-x: hidden;
}

.item-grid {
  display: flex;
  flex-direction: row;
}

.item-img {
  height: 230px;
  box-shadow: 0 6px 5px 0 rgba(0, 0, 0, 0.5);
}

.res-image {
  height: 100% !important;
  width: auto;
}

.change-game-block {
  padding: 6rem 2rem;
}


.change-game-block span {
  padding-bottom: 10px;
  border-bottom: 1px solid #000;
  color: #000;
  font-size: 17px;
  font-weight: 500;
}

.change-game-block span:hover {
  text-decoration: none;
  border-bottom: 1px solid #468ce2;
  color: #468ce2;
  cursor: pointer;
}