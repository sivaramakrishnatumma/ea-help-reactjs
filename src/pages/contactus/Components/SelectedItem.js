import React from "react";

import "./SelectedItem.css";
import Image from "../../../shared/components/UIElements/Image/Image";

const SelectedItem = props => {
  const { name, imageUrl, type, onChange } = props;

  return (
    <React.Fragment>
      <div className="selected-item-container">
        <div className="item-grid">
          <div className="item-img">
            <Image src={imageUrl} alt={name} className="res-image" />
          </div>
          <div className="change-game-block" onClick={onChange}>
            <a>Change {type}</a>
          </div>
        </div>
      </div>
    </React.Fragment>
  );
};

export default SelectedItem;
