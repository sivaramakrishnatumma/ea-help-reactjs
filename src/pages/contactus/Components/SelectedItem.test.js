import React from "react";
import { mount } from "enzyme";
import SelectedItem from "./SelectedItem";

describe("SelectedItem component test case", () => {
  it("should test SelectedItem component with default props", () => {
    const props = {
      name: "test name",
      imageUrl: "test-url",
      type: "test",
      onChange: jest.fn()
    };
    const wrapper = mount(<SelectedItem {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
