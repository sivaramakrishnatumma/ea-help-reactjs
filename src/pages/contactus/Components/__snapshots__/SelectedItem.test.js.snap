// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedItem component test case should test SelectedItem component with default props 1`] = `
<SelectedItem
  imageUrl="test-url"
  name="test name"
  onChange={[MockFunction]}
  type="test"
>
  <div
    className="selected-item-container"
  >
    <div
      className="item-grid"
    >
      <div
        className="item-img"
      >
        <Image
          alt="test name"
          className="res-image"
          src="test-url"
        >
          <img
            className="img-fluid res-image"
            src="test-url"
          />
        </Image>
      </div>
      <div
        className="change-game-block"
        onClick={[MockFunction]}
      >
        <span>
          Change 
          test
        </span>
      </div>
    </div>
  </div>
</SelectedItem>
`;
